### PR TITLE
adding button block starting point

### LINKED ISSUE #2873: Create button block
# Story

As a site owner, I can create a button block component so that I don't have to use the CTA block component.

# Proposed solution

Create a button block which implements the markup and options from the UIDS 4 button component with these options:

Link field with a title. 

- Button type: 
  - Primary 
  - Secondary
  - Tertiary
- Transparent: true
- Button width: full-width
- Borderless: true
- Font: thin - Antonio is the bold default, Roboto is the secondary thin option. 
- Font awesome icon selector (default to arrow for first slice)
- Limit button link text to a certain number of characters?
- Button size: s, m, l - unclear if we want to release this option but worth exploring for the CTA row.

### PR DESCRIPTION
**Pull request by GaryRidgway**
Tracks https://github.com/uiowa/uiowa/issues/2873

##To test
Check out branch
Sync a site
Make a button block in Layout builder
Try all the styles
Profit????

### CODE REVIEW

File: docroot/modules/custom/layout_builder_custom/layout_builder_custom.module [hunk 1]
```diff
@@ -794,6 +797,12 @@ function _layout_builder_custom_process_element(array $element, FormStateInterfa
 
       break;
 
+    case 'inline_block:uiowa_button':
+      // Add some things here.
+      $element["field_uiowa_button_link"]["widget"][0]["title"]["#maxlength"] = 25;
+      $foo = 'bar';
```
Comment (bspeare) [inline docroot/modules/custom/layout_builder_custom/layout_builder_custom.module]: Remove `$foo` here?

Comment (GaryRidgway) [reply docroot/modules/custom/layout_builder_custom/layout_builder_custom.module]: not anymore!

**Comment by bspeare** [comment]
Something like https://github.com/uiowa/uiowa/issues/2670 would be useful for these blocks to prevent stacking. 

Screen Shot 2022-08-01 at 2 44 01 PM

File: docroot/modules/custom/layout_builder_custom/layout_builder_custom.module [hunk 2]
```diff
@@ -794,6 +797,10 @@ function _layout_builder_custom_process_element(array $element, FormStateInterfa
 
       break;
 
+    case 'inline_block:uiowa_button':
+      $element["field_uiowa_button_link"]["widget"][0]["title"]["#maxlength"] = 25;
```
Comment (pyrello) [inline docroot/modules/custom/layout_builder_custom/layout_builder_custom.module]: Please change to single quotes :)

File: docroot/themes/custom/uids_base/templates/uids/button.html.twig [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{% set attributes = attributes ?: create_attribute() %}
+<div {{ attributes }}>
+  {% block button_content_start %}
+    {{ title_prefix }}
+    <a {{ button_classes }} href="{{ button_link_url }}">
+      {{ button_link_title }}
```
Comment (pyrello) [inline docroot/themes/custom/uids_base/templates/uids/button.html.twig]: I'd be interested in moving away from having `button_`-prefixed variables names, at least in the context of this file.

File: docroot/modules/custom/layout_builder_custom/layout_builder_custom.module (hunk 2 above)
Comment (GaryRidgway) [reply docroot/modules/custom/layout_builder_custom/layout_builder_custom.module]: can do! I'm sorry lol, when i cope th path from PHPStorm it does doubles!

File: docroot/themes/custom/uids_base/templates/uids/button.html.twig (hunk 3 above)
Comment (GaryRidgway) [reply docroot/themes/custom/uids_base/templates/uids/button.html.twig]: I will try and refactor!

Comment (GaryRidgway) [reply docroot/themes/custom/uids_base/templates/uids/button.html.twig]: This hsa been achieved!

**Comment by GaryRidgway** [comment]
> Something like #2670 would be useful for these blocks to prevent stacking.
> 
> Screen Shot 2022-08-01 at 2 44 01 PM

I totally agree, any opinions on proceeding? probing now? requesting sprint prioritization? waiting and asking?

**Comment by bspeare** [comment]
I added a `block` class to the wrapping `div` on [this ](https://github.com/uiowa/uiowa/pull/5500/commits/e05c3f818167fe364af61a5707d5bf339451b437) commit so that it gets this property applied to it: https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/scss/global.scss#L40.  Ideally, we should just pick a class we can use for all of our blocks so that we can reduce this line of code and use it in future work like https://github.com/uiowa/uiowa/issues/2670#issuecomment-1201694625.

**Comment by makurj** [comment]
Pulling down to test.

**Comment by makurj** [comment]
Should I test all content blocks with all button styles, or 1 content block with all button styles?

**Comment by GaryRidgway** [comment]
> Should I test all content blocks with all button styles, or 1 content block with all button styles?


Just the button block
Add block > button

**Review by makurj — APPROVED** [review]
Seems to work as expected. Tried all styles with a few different background and section options.

**Pull request merged**